lib/routes/signin-codes.js

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const isA = require('joi')
+const validators = require('./validators')
+
+const METRICS_CONTEXT_SCHEMA = require('../metrics/context').requiredSchema
+
+module.exports = (log, db, config, customs) => {
+  const CODE_LENGTH = Math.ceil(4 * (config.signinCodeSize / 3))
+
+  return [
+    {
+      method: 'POST',
+      path: '/signinCodes/consume',
+      config: {
+        validate: {
+          payload: {
+            code: isA.string().regex(validators.URL_SAFE_BASE_64).length(CODE_LENGTH).required(),
```

Comment:
Not sure if it qualifies as "harm" but it would mean that changing the code size is not just a config change, we'd need to change the source code and tag a new release.